### DISCUSSION TITLE
re-org: protocol, user support libs, ...

### DISCUSSION THREAD

**Discussion by marcellanz** [eigr/massa#61] [Ideas]
I'd like to re-org certain things we inherit from Cloudstate. To get a container with all things Cloustate and Akkaserverless, I've already forked every repository that I found relevant into https://github.com/eigr-labs. My plan would be to stay in sync with their stuff and not having them in our main repo.

Next step would be to establish our own protocol and user support libraries within eigr. That would be a fork of the existing work we did with LB and then re-branded and re-named for our project. 

I'm still not sure if I'd use eigr as the main name, and therefore the Go User Language Support lib is "eigr/go-support" or if we have something like "eigr/serverless-go-support" or "eigr/whatever-go-support". You might found out that eigr, like akka, is a mountain :). We don't have to hold that 1:1. I found eigr a nice name to contain certain stuff. So "eigr/protocol" is nice... or then  eigr/perma for our state models. I'm not sure, lets find out.

Tasks I see:
- [ ] establish our own protocol
- [ ] user support libraries within eigr re-branded and re-named for our project
- [ ] decide with what naming we go forward with "user support library" vs SDKs vs. ?

**Comment by sleipnir** [comment]
>     * [ ]  decide with what naming we go forward with "user support library" vs SDKs vs. ?

I think using user support library or sdk makes the same thing, both are interchangeable terms. I think in our documentation we should have a glossary of terms and that's where we would explain that both terms can be found and that they have the same meaning.

>     * [ ]  user support libraries within eigr re-branded and re-named for our project

I agree. But I would like to rewrite these sdks so that it is easier to evolve in the future with no dependencies than what has been done in the past. We can even fork what exists, but we should create a new branch and work with a blank slate. Obviously it will be possible to bring in some of the existing code if necessary.

>     * [ ]  establish our own protocol

Agree. But there is an outstanding question about the question we asked Lightbend. This directly affects how we will do our own protocol.

**Reply by marcellanz** [reply]
> Agree. But there is an outstanding question about the question we asked Lightbend. This directly affects how we will do our own protocol.

It is true, but our proposal was for the discovery service to start with only. Also I think we will go with the Cloudstate proto files, and evolve. Even if LB agrees on our proposal, they will not reverse much of their changes to their proto files. The discovery protocol they already changed a bit. Uppercase method names, new method.

**Reply by marcellanz** [reply]
look at the current state of their discovery protocol:
https://github.com/eigr-labs/akkaserverless-protocol/blob/main/akkaserverless-proxy-protocol-0.7.0-beta.10/akkaserverless/protocol/discovery.proto

there is not much I would disagree on. I can accept these changes; the proposal we sent for me is also to let them have a last chance to get on common ground. If they ignore or refuse, I have no hesistation to ignore them forever. They did not care for a long time now, even ignored our critisism.

**Reply by sleipnir** [reply]
I understand perfectly.
So I think it's good that we already changed our discovery service to what is best for us, we can adopt what was done in the new LB service and add the attributes that address the issue of the Implementation Provider, so we would already have a technical proposal to present to them.
